### DISCUSSION TITLE
Viewport tests failing in headless Chrome 130.0.6723.69.

### DISCUSSION THREAD

**Discussion by AfafRH19** [teamcapybara/capybara#2787] [Q&A]
Hey everyone!
I’m encountering some issues with the viewport since upgrading WebDriver to version 130.0.6723.69.
I have an integration test with a component that has the class "map-container." We used the following function to check if the component is within the viewport:
def in_viewport?(element, top_margin = 0)
  evaluate_script(
    "(function(el) {
      var rect = el.getBoundingClientRect();
      return (
        rect.top >= #{top_margin} && rect.left >= 0 &&
        rect.bottom <= (window.innerHeight || document.documentElement.clientHeight) &&
        rect.right <= (window.innerWidth || document.documentElement.clientWidth)
      );
    })(arguments[0]);",
    element
  )
end
However, since the recent upgrade, this check always returns false when running in HEADLESS mode. Any insights or suggestions on how to resolve this?

**Comment by twalpole** [comment]
Sounds like a driver behavior change. Since this is all via JS there's really nothing Capybara can do about it
